realm/realm-library/src/androidTest/java/io/realm/DynamicRealmObjectTests.java

```diff
@@ -1629,4 +1631,95 @@ public void run() {
         thread.start();
         TestHelper.awaitOrFail(threadFinished);
     }
+
+    @Test
+    public void getNullableFields() {
+        realm.executeTransaction(realm -> {
+            NullablePrimitiveFields primitiveNullables = realm.createObject(NullablePrimitiveFields.class);
+            primitiveNullables.setFieldBoolean(null);
+
+            assertNull(primitiveNullables.getFieldBoolean());
+            assertNull(primitiveNullables.getFieldInt());
+            assertNull(primitiveNullables.getFieldFloat());
+            assertNull(primitiveNullables.getFieldDouble());
+            assertNull(primitiveNullables.getFieldString());
+            assertNull(primitiveNullables.getFieldBinary());
+            assertNull(primitiveNullables.getFieldDate());
+
+            realm.delete(AllJavaTypes.class);
+            AllJavaTypes allJavaTypes = realm.createObject(AllJavaTypes.class, UUID.randomUUID().getLeastSignificantBits());
+
+            assertNull(allJavaTypes.getFieldObject());
+            allJavaTypes.getFieldBooleanList().add(null);
```

Comment:
These tests do not seem related in a test named `getNullableFields` Lists can never be `null` and I hope that adding null is tested elsewhere, if not, it should probably be in its own test?

Reply:
The issue was that we could not return null values for nullable fields from a dynamic realm. I thought it would be safest to also included a check of retrieving a null value from a list from a dynamic realm. I could try to find another place for the test...or update the name of the test to `getNullForNullables` ... ?